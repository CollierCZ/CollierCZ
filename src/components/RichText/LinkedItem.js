import { css } from "@emotion/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faQuoteLeft, faQuoteRight } from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import React from "react";

const LinkedItem = ({ linkedItem }) => {
  const type = linkedItem.system.type;

  switch (type) {
    case "quote": {
      const source = linkedItem.elements.source.resolvedHtml
        ? `<footer>${linkedItem.elements.source.resolvedHtml}</footer>`
        : "";
      const quote = source
        ? linkedItem.elements.quote.resolvedHtml + source
        : linkedItem.elements.quote.resolvedHtml;
      return (
        <div
          css={css`
            display: flex;
            svg {
              font-size: 5rem;
            }
            footer {
              font-size: 2rem;
            }
          `}
        >
          <FontAwesomeIcon icon={faQuoteLeft} />
          <blockquote
            dangerouslySetInnerHTML={{ __html: quote }}
            css={css`
              font-style: normal;
              margin: 0 3rem;
            `}
          />
          <FontAwesomeIcon icon={faQuoteRight} />
        </div>
      );
    }

    default:
      return null;
  }
};

LinkedItem.propTypes = {
  linkedItem: PropTypes.object.isRequired
};

export default LinkedItem;
