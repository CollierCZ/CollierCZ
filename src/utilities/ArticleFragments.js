import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`