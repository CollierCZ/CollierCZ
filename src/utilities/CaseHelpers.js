const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}