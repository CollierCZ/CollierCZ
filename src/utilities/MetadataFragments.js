import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`