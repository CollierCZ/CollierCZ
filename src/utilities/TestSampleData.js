export const author = {
  short_bio: {
    value:
      '<p>Originally from <a href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts" data-new-window="true" target="_blank" rel="noopener noreferrer">Lexington</a> and now residing in <a href="https://www.gotobrno.cz/en/" data-new-window="true" target="_blank" rel="noopener noreferrer">Brno</a>, Aaron is an educator, editor, and elucidator.</p>'
  },
  picture: {
    value: [
      {
        url:
          "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/f7377b84-8ff0-4b77-bc0f-cfdc34d379cc/quixote.jpg"
      }
    ]
  },
  name: {
    value: "Aaron Collier"
  },
  url: {
    value: "aaron-collier"
  }
};

export const simpleImage = {
  description: "A cool picture",
  url: "https://example.com/"
};

export const sampleQuote = {
  system: {
    type: "quote"
  },
  elements: {
    quote: {
      resolvedData: {
        html:
          "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
      }
    },
    source: {
      resolvedData: {
        html:
          "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
      }
    }
  }
};

export const sampleShowcase = {
  system: {
    type: "showcase"
  },
  elements: {
    items: {
      linked_items: [
        {
          id: "457d3938-5907-5cca-947d-bae1583e105d",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=vj7TyUgBEW8"
            },
            name: {
              value: "Write the Docs talk"
            },
            short_description: {
              links: [
                {
                  linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
                  urlSlug: "seeing-your-docs-through-different-eyes",
                  type: "article",
                  codename: "write_the_docs_2019"
                }
              ],
              resolvedData: {
                html:
                  '<p>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference. Or read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a>.</p>'
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG"
                }
              ]
            }
          },
          fields: {
            slug: "write-the-docs-talk"
          }
        },
        {
          id: "fce4fe5b-59c9-5c88-8634-16b63df49d4a",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=MlMkAqxoPZM"
            },
            name: {
              value: "Structured content video"
            },
            short_description: {
              links: [],
              resolvedData: {
                html:
                  "<p>A video about what structured content is and how it can benefit content creators.</p>"
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg"
                }
              ]
            }
          },
          fields: {
            slug: "structured-content-video"
          }
        }
      ]
    }
  }
};
