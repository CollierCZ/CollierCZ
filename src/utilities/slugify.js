let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify