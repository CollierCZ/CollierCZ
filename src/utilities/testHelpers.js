import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
