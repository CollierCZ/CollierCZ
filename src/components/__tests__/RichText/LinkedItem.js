import React from "react";
import LinkedItem from "../../RichText/LinkedItem";
import { sampleQuote, sampleShowcase } from "../../../utilities/TestSampleData"
import { simpleRenderer } from "../../../utilities/testHelpers";

const unmatchedType = {
  system: {
    type: 'wacky'
  }
}

describe("LinkedItem", () => {
  it("renders a quote correctly", () => {
    simpleRenderer(<LinkedItem linkedItem={sampleQuote} />);
  });
  it("renders a showcase correctly", () => {
    simpleRenderer(<LinkedItem linkedItem={sampleShowcase} />);
  });
  it("renders null when no match", () => {
    simpleRenderer(<LinkedItem linkedItem={unmatchedType} />);
  });
});

