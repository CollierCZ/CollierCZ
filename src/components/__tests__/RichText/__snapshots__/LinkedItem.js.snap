// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedItem renders a quote correctly 1`] = `
.c0 {
  width: 32px;
  height: 32px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #00A991;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #5F738C;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #00A991;
}

<div
  css={
    Array [
      "
        display: flex;
        align-items: center;
      ",
    ]
  }
>
  <svg
    className="c0"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 24 24"
  >
    <path
      d="M20.696 4.8H3.304C2.584 4.8 2 5.384 2 6.104V17.89a1.269 1.269 0 0 0 1.304 1.259h2.609v3.478a.435.435 0 0 0 .683.357l5.54-3.835h8.56c.72 0 1.304-.584 1.304-1.304V6.103c0-.72-.584-1.304-1.304-1.304zm-2.609 7.341a1.304 1.304 0 1 1-2.609 0 1.304 1.304 0 0 1 2.609 0zm-12.174 0a1.304 1.304 0 1 1 2.609 0 1.304 1.304 0 0 1-2.609 0zm4.783 0a1.304 1.304 0 1 1 2.608 0 1.304 1.304 0 0 1-2.608 0z"
    />
  </svg>
  <blockquote>
    <p
      className="c1"
    >
      You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.
    </p>
    <footer>
      <p
        className="c2"
      >
        <p>
          Atticus Finch in 
          <cite>
            To Kill a Mockingbird 
          </cite>
          by Harper Lee
        </p>
      </p>
    </footer>
  </blockquote>
</div>
`;

exports[`LinkedItem renders a showcase correctly 1`] = `
.c6 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c6,
.c6:link,
.c6:visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c6:hover,
.c6:active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #00A991;
}

.c6:hover .TextLink__IconContainer-sc-1bvlje4-0,
.c6:active .TextLink__IconContainer-sc-1bvlje4-0 {
  color: #00A991;
}

.c6:focus {
  outline: none;
  background: rgba(0,169,145,0.1);
  box-shadow: 0 0 0 4px rgba(0,169,145,0.1);
}

.c6:focus:not(:focus-visible) {
  background: rgba(0,169,145,0.1);
  box-shadow: 0 0 0 4px rgba(0,169,145,0.1);
}

.c6:-moz-focusring,
.c6:focus-visible {
  background: rgba(0,169,145,0.1);
  box-shadow: 0 0 0 4px rgba(0,169,145,0.1);
}

.c4 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c4 a:not(.c5),
.c4 a:not(.c5):link,
.c4 a:not(.c5):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c4 a:not(.c5):hover,
.c4 a:not(.c5):active {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #00A991;
}

.c3 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 28px;
  font-weight: 700;
  color: #252A31;
  line-height: 36px;
  margin: 0;
}

.c2 {
  width: 100%;
}

.c2 > * {
  margin: 0 0 16px 0!important;
}

.c2 > *:last-child {
  margin: 0 !important;
}

.c1 {
  font-size: 14px;
  line-height: 20px;
  cursor: pointer;
  padding: 16px;
}

.c0 {
  display: block;
  width: 100%;
  box-sizing: border-box;
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  color: #252A31;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #FFFFFF;
  border-radius: 3px;
  box-shadow: 0px 0px 2px 0px rgba(37,42,49,0.16),0px 1px 4px 0px rgba(37,42,49,0.12);
  -webkit-transition: box-shadow 0.15s ease-in-out;
  transition: box-shadow 0.15s ease-in-out;
}

.c0:hover,
.c0:focus {
  outline: none;
  box-shadow: 0px 1px 4px 0px rgba(37,42,49,0.16),0px 4px 8px 0px rgba(37,42,49,0.12);
}

.c0:hover .TileHeader__StyledIconRight-sc-13kgyxb-4,
.c0:focus .TileHeader__StyledIconRight-sc-13kgyxb-4 {
  color: #52647A;
}

@media (min-width:414px) {

}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {
  .c2 > * {
    margin: 0 0 16px 0!important;
  }
}

@media (min-width:1200px) {

}

@media (min-width:576px) {
  .c1 {
    padding: 24px;
  }
}

<div
  css={
    Array [
      "
      display: flex;
      flex-wrap: wrap;
    ",
    ]
  }
>
  <div
    css={
      Array [
        "
        flex: 1 1 34ch;
        margin: 16px;
        > * {
          width: 100%;
          height: 100%;
        }
      ",
      ]
    }
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <div
            className="c3"
          >
            Write the Docs talk
          </div>
          <div
            css={
              Array [
                "
              background-size: cover;
              width: 100%;
              height: 200px;
              background-image: url(",
                "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG",
                ");
            ",
              ]
            }
          />
          <p
            className="c4"
          >
            A video of a talk I gave at the 
            <a
              className="c5 c6"
              href="http://www.writethedocs.org/conf/prague/2019/"
              onClick={[Function]}
              rel=""
            >
              Write the Docs Prague 2019
            </a>
             conference. Or read a 
            <a
              className="c5 c6"
              href="/articles/seeing-your-docs-through-different-eyes"
              onClick={[Function]}
              rel=""
            >
              text version of the talk
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    css={
      Array [
        "
        flex: 1 1 34ch;
        margin: 16px;
        > * {
          width: 100%;
          height: 100%;
        }
      ",
      ]
    }
  >
    <div
      className="c0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <div
            className="c3"
          >
            Structured content video
          </div>
          <div
            css={
              Array [
                "
              background-size: cover;
              width: 100%;
              height: 200px;
              background-image: url(",
                "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg",
                ");
            ",
              ]
            }
          />
          <p
            className="c4"
          >
            A video about what structured content is and how it can benefit content creators.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LinkedItem renders null when no match 1`] = `null`;
